### PR TITLE
Adding option to provide pom/gradle path for Azure Test Plan task

### PR DESCRIPTION
**Pull request by KathanS**
**Task name**: AzureTestPlan

**Description**: Adding option to provide pom/gradle path for Azure Test Plan task

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

**Testing**

**Maven**

Project structure

We can see that pom.xml file is not at the root of the repo but inside Project directory

image

Without option to provide file path

image

With option to provide file path

image

**Gradle**

Project structure

We can see that gradle - build files is not at the root of the repo but inside Project directory

image

Without option to provide file path

image

With option to provide file path

image

ATP-UI Demo

On UI, for selecting language framework users will be shown Java-Maven (with -) and internally we will be receiving value as JavaMaven (ref: due to issues with special chars in visibleRule: _ or #, ref: https://github.com/microsoft/azure-pipelines-task-lib/issues/658)

ATP-UI2

**Demo without providing the file path incase of default project structures** (working fine as before)

**Maven**

Project structure

image

Pipeline

image

**Gradle**

Project Structure

image

Pipeline

image

### CODE REVIEW

File: Tasks/AzureTestPlanV0/task.json [hunk 1]
```diff
@@ -77,13 +77,31 @@
...
         "Jest": "Jest"
       }
     },
+    {
+      "name": "pomFilePath",
+      "type": "string",
+      "label": "Pom file path",
+      "defaultValue": "",
+      "required": false,
+      "helpMarkDown": "Add custom file path of pom file incase it is not pom.xml at root directory",
```
Comment (adityashahms) [inline Tasks/AzureTestPlanV0/task.json:93]: Let's have the description as "Relative path from the repository root to the Maven POM file." 

This ensure consistent experience across all the tasks. For ref, maven: [Link](https://github.com/microsoft/azure-pipelines-tasks/blob/18b2a6e45201b869e5439f15ea32129602b3babf/Tasks/MavenV4/task.json#L62C5-L72C7) and gradle: [Link](https://github.com/microsoft/azure-pipelines-tasks/blob/18b2a6e45201b869e5439f15ea32129602b3babf/Tasks/GradleV3/task.json#L48C5-L58C7)

File: Tasks/AzureTestPlanV0/task.json [hunk 2]
```diff
@@ -77,13 +77,31 @@
...
+      "visibleRule": "testLanguageInput = JavaMaven"
+    },
+    {
+      "name": "gradleFilePath",
+      "type": "string",
+      "label": "Gradle file path",
+      "defaultValue": "",
+      "required": false,
+      "helpMarkDown": "Add custom file path of gradle build file incase it is not build.gradle at root directory",
```
Comment (adityashahms) [inline Tasks/AzureTestPlanV0/task.json:102]: Update this helpMarkdown text as well.

File: Tasks/AzureTestPlanV0/automatedTestInvoker.ts [hunk 3]
```diff
@@ -19,14 +21,14 @@ export async function testInvoker(testsToBeExecuted: string[], ciData: ciDiction
         }
 
         switch (testLanguage) {
-            case 'Java-Maven':
-                exitCode = await executeMavenTests(testsToBeExecuted);
+            case 'JavaMaven':
```
Comment (adityashahms) [inline Tasks/AzureTestPlanV0/automatedTestInvoker.ts:24]: Add the testing screenshot that without the pom file path provided, we are by default looking at the root of the repository for the pom.xml file.

**Review by adityashahms — CHANGES_REQUESTED** [review]

File: Tasks/AzureTestPlanV0/task.json (hunk 1 above)
Comment (KathanS) [reply Tasks/AzureTestPlanV0/task.json]: Updated help markdown text for maven

File: Tasks/AzureTestPlanV0/task.json (hunk 2 above)
Comment (KathanS) [reply Tasks/AzureTestPlanV0/task.json]: Updated help markdown text for gradle

File: Tasks/AzureTestPlanV0/automatedTestInvoker.ts (hunk 3 above)
Comment (KathanS) [reply Tasks/AzureTestPlanV0/automatedTestInvoker.ts:24]: Updated description to add testing screenshots for default project structure

**Review by adityashahms — APPROVED** [review]
LGTM

**Pull request merged**